File: easy/3550-smallest-index-with-digit-sum-equal-to-index/solution.cpp
class Solution {
public:
    int smallestIndex(vector<int>& nums) {
        for(int i{0}; i < nums.size(); ++i){
            int index{i}, value{nums[i]};
            while(value > 0){
                index -= value % 10;
                value /= 10;
            }
            if(index == 0)
                return i;
        }
        return -1;
    }
};

File: easy/3550-smallest-index-with-digit-sum-equal-to-index/solution.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        for i in range(0,len(nums)):
            val, index = nums[i], i
            while val > 0 :
                index -= val%10
                val //=10
            if index == 0 : 
                return i
        return -1
